web/packages/shared/components/UnifiedResources/UnifiedResources.tsx

```diff
@@ -388,9 +385,21 @@ export function UnifiedResources(props: UnifiedResourcesProps) {
   );
 }
 
-export function generateResourceKey({
-  resource,
-}: SharedUnifiedResource): string {
+function getResourcePinningSupport(
+  pinningNotSupported: boolean,
+  updatePinnedResourcesAttempt: Attempt<void>
+): PinningSupport {
```

Comment:
This never returns the new enum `PinningSupport.Hidden`. Perhaps we add it as an argument after `pinningNotSupported` ?

Reply:
Good point, added.

However, while adding it, I realized that we can improve the props passed to `UnifiedResources` a bit.
If we don't support pinning, or just want to hide it, there is no point in passing function to get and update them.
Also, when pinning is not supported, we shouldn't even try to fetch them :/

I refactored all these props to an object that requires get/update functions only when pinning is supported.

Reply:
Oh yeah, much better